flagext/secret_test.go

```diff
@@ -75,3 +76,58 @@ func TestSecretdYAML(t *testing.T) {
 		assert.Equal(t, testStruct, actualStruct)
 	}
 }
+
+func TestSecretEquals(t *testing.T) {
```

Comment:
[Nit] I personally prefer the test naming convention `Test$TYPE_$METHOD`, makes it easier to understand what's being tested:
```suggestion
func TestSecret_Equals(t *testing.T) {
```

Reply:
SGTM!